src/SwedbankPay.Sdk/ResourceBase.cs

```diff
@@ -0,0 +1,71 @@
+﻿namespace SwedbankPay.Sdk
+{
+    using RestSharp;
+    using RestSharp.Authenticators;
+
+    using SwedbankPay.Sdk.Exceptions;
+    using SwedbankPay.Sdk.PaymentOrders;
+
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+
+    public abstract class ResourceBase
+    {
+        protected readonly SwedbankPayOptions SwedbankPayOptions;
+        private readonly ILogSwedbankPayHttpResponse _logger;
+
+        internal ResourceBase(SwedbankPayOptions swedbankPayOptions, ILogSwedbankPayHttpResponse logger)
+        {
+            SwedbankPayOptions = swedbankPayOptions;
```

Comment:
What happens if `swedbankPayOptions` is `null`? All constructors need proper argument guards.

```suggestion
            SwedbankPayOptions = swedbankPayOptions ?? throw new ArgumentNullException(nameof(swedbankPayOptions));
```